client/src/pages/GrantAccessForm.js

```diff
@@ -0,0 +1,213 @@
+import { useState } from "react";
+import "./GrantAccessForm.css";
+import { post } from "../TenderClient";
+import Logo from "../assets/images/CTY-logo-rectangle.png";
+
+const GrantAccessForm = () => {
+	const [role, setRole] = useState("bidder");
+	const [buyerDetails, setBuyerDetails] = useState({
+		userType: "buyer",
+		company: "",
+		description: "",
+		address: "",
+		email: "",
+	});
+
+	const [bidderDetails, setBidderDetails] = useState({
+		userType: "bidder",
+		firstName: "",
+		lastName: "",
+		email: "",
+	});
+
+	const [resgisterStatus, setRegisterStatus] = useState("");
+
+	async function postBuyerDetails(buyerData) {
+		try {
+			const data = await post("/api/signup", buyerData);
+			if (data) {
+				setRegisterStatus("Successfully registered.");
+			} else {
+				setRegisterStatus("Registration failed!");
+			}
+		} catch (error) {
+			setRegisterStatus("Validation Error");
+		}
```

Comment:
This is basically the same code as `postBidderDetails`. Extract this logic to a function and reuse it.